_dev/back/pages/Logs.vue

```diff
@@ -0,0 +1,134 @@
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ *-->
+<template>
+  <div class="panel">
+    <div class="panel-buttons">
+      <button
+        type="button"
+        class="btn btn-primary"
+        @click.prevent="refreshLogs()"
+      >
+        REFRESH LOGS
+        <i class="material-icons">refresh</i>
+      </button>
+
+      <button
+        class="btn"
+        :class="{'btn-success': !isJSEventListenerIsEnabled(), 'btn-warning': isJSEventListenerIsEnabled()}"
+        @click.prevent="toggleJSEventListener()"
+      >
+        <template v-if="isJSEventListenerIsEnabled()">
+          Disable JS Events Listener
+        </template>
+        <template v-else>
+          Enable JS Events Listener
+        </template>
+      </button>
+    </div>
+
+    <div class="alert alert-warning" role="alert">
+      Module PS Quality Assurance is hooked on hook 'displayTop' (alias 'header')
+      so the logs related to this hook are not displayed here.
+    </div>
+
+    <table
+      class="table table-striped"
+      v-for="(logs, requestIdentifier) in logsGroupedByRequest"
+      :key="requestIdentifier"
+    >
+      <thead>
+        <tr>
+          <th>#</th>
+          <th>Hook</th>
+          <th>Parameters</th>
+          <th>Output</th>
+          <th>Time</th>
+          <th>Status</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr
+          v-for="log in logs"
+          :key="log.id"
+          :class="{ 'table-danger': (log.error === '1') }"
+        >
+          <th scope="row">{{ log.id }}</th>
+          <td>{{ log.hook_name }}</td>
+          <td>{{ log.hook_parameters }}</td>
+          <td>{{ log.output }}</td>
+          <td>{{ log.called_at }}</td>
+          <td>
+            <i class="material-icons">
+              <template v-if="log.error === '0'">
+                check
+              </template>
+              <template v-else>
+                clear
+              </template>
+            </i>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script>
+  import api from '@/lib/api';
+
+  export default {
+    name: 'Logs',
+    data() {
+      return {
+        jsEventListenerIsEnabled: false,
+        logsGroupedByRequest: [],
+      };
+    },
+    mounted() {
+      this.refreshLogs();
+      this.refreshJSEventListenerStatus();
+    },
+    methods: {
+      refreshLogs() {
+        api.getHookCallLogs().then((res) => {
+          this.logsGroupedByRequest = res.data;
+        });
+      },
+      refreshJSEventListenerStatus() {
+        api.refreshJSEventListenerStatus().then((res) => {
+          this.jsEventListenerIsEnabled = res.data;
+        });
+      },
+      toggleJSEventListener() {
+        api.toggleJSEventListener().then(() => {
+          this.refreshJSEventListenerStatus();
+        });
+      },
+      isJSEventListenerIsEnabled() {
+        return this.jsEventListenerIsEnabled;
+      },
```

Comment:
```suggestion
```

Reply:
So you're against getter for property of the class itself?

Reply:
This is overengineering, you can easily check the state itself instead of using a method/computed

Reply:
Remember this is not PHP, this is JS, which globally aim a functional programming

Reply:
I'm a backend PHP developer writing JS, what did you expect? 😆

Reply:
Data elements are already objects.
Use data if you don't have to do special things on it.
Use methods to call a function when an event happens in the DOM or to call a function from the computed or watchers when something happens in your component or when you need to pass parameters
Use Computed properties when you need to compose new data from existing data sources or you need to reference a value directly in your template

Reply:
👍